### PR TITLE
initial draft of q3

### PR DESCRIPTION
**Pull request by BooleanJulien**

### CODE REVIEW

File: src/gapminder_challenge/dashboard/dash_app3.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            # only show the line for selected filter region
+            df_q3 = df[df['sub_region']==filter]
+            df_q3 = df_q3.groupby(['sub_region', 'year']).mean()
+            df_q3 = df_q3.reset_index()  
+
+            chart = alt.Chart(df_q3.query(f'year>={year_range_slider[0]} and year<={year_range_slider[1]}'), 
+                title="Average Life Expectancy").mark_line().encode(
+                y=alt.Y("life_expectancy", title="Average Life Expectancy (Years)"),
+                x=alt.X("year", title="Year"),
+                color='sub_region',
```
Comment (SonQBChau) [inline src/gapminder_challenge/dashboard/dash_app3.py:70]:
```suggestion
                color=alt.Color('sub_region', legend=None),
```

File: src/gapminder_challenge/dashboard/dash_app3.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        :param year_range_slider: The year range to plot
+        :param filter_dropdown: The filter to plot
+        :return: The Altair chart is being returned.
+        """
+        filter = filter_dropdown
+        if filter=="All" or filter=="":
+            df_q3 = df.groupby(['sub_region', 'year']).mean()
+            df_q3 = df_q3.reset_index()
+            chart = alt.Chart(df_q3.query(f'year>={year_range_slider[0]} and year<={year_range_slider[1]}'), 
+                title="Average Life Expectancy").mark_line().encode(
```
Comment (SonQBChau) [inline src/gapminder_challenge/dashboard/dash_app3.py:55]:
```suggestion
                title=f"Average Life Expectancy from {year_range_slider[0]} to {year_range_slider[1]}").mark_line().encode(
```

**Review by SonQBChau — APPROVED** [review]
Look great! I just suggest a small change since your legend only display one at a time, so it's not needed and we should move it to the title

**Comment by BooleanJulien** [comment]
Fixed merge conflicts, thanks for the changes @SonQBChau

**Pull request merged**